components/date-picker/internal/DatePickerBase.cs

```diff
@@ -783,7 +804,7 @@ public virtual void ClearValue(int index = 0, bool closeDropdown = true)
             return null;
         }
 
-        public void InvokeStateHasChanged()
+        public new void InvokeStateHasChanged()
```

Comment:
We can change this method to `virtual`

Reply:
Hey @ElderJames.  This will require changing the implementation in the `AntComponentBase` to `public virtual` from `protected` for `IDatePicker` interface to work. And, in this case, there is probably no need for implementation in the [DatePickerBase.cs](https://github.com/ant-design-blazor/ant-design-blazor/pull/2512/files/d3a808e4c8e106f751d86623e4749b4ee380c51e#diff-dbbc0da12dd4220386a71c6ed7f2afa809932e99e6f44740f3b3e10cd73493f6)

Reply:
> We can change this in AntComponentBase to `protected virtual`, right?
Yes, we can, but it's `public `in `DatePickerBase`, we can't override it.

Reply:
It's called by `DatePickerTemplate` via `IDatePicker` interface, `protected `will break this.

Reply:
It's a little confusing.

Reply:
> It's a little confusing.

Yes. This was not designed by me. I've added a `new` modifier because of a compiler warning about the method which hides an inherited one.

Reply:
I've fixed this by removing the `InvokeStateHasChanged` from `IDatePicker` altogether. There is no need for it since it's called a few lines lower in `DatePickerTemplate` when the placeholder value is set. So basically, it was invoked multiple times, causing unnecessary renderings. 

Also, when the Value is selected, there is no need to set the placeholder on mouse hover and call StateHasChanged again? I've fixed this too.

Reply:
That's great!